File: RomansShop.Domain/Entities/Order.cs
using System;
using System.Collections.Generic;
using RomansShop.Domain.Extensibility;

namespace RomansShop.Domain.Entities
{
    public class Order : IEntity
    {
        public Guid Id { get; set; }

        public Guid? UserId { get; set; }

        public string CustomerEmail { get; set; }

        public string CustomerName { get; set; }

        public string Address { get; set; }

        public decimal Price { get; set; }

        public DateTime Date { get; set; } = DateTime.Now;

        public OrderStatus Status { get; set; }

        public IList<OrderProduct> OrderProducts { get; set; } = new List<OrderProduct>();
    }
}

File: RomansShop.ClientApp/src/app/shared/order.ts
import { Product } from "./product";
import { OrderStatus } from "./order-status";

export class Order {
    constructor(public id?: string,
                public userId?: string,
                public customerEmail?: string,
                public customerName?: string,
                public products: Product[] = new Array<Product>(),
                public address?: string,
                public price?: number,
                public status?: OrderStatus) {
    }
}